CrossrefReferenceLinkingPlugin.php

```diff
@@ -0,0 +1,474 @@
+<?php
+
+/**
+ * @file plugins/generic/crossrefReferenceLinking/CrossrefReferenceLinkingPlugin.inc.php
+ *
+ * Copyright (c) 2013-2023 Simon Fraser University
+ * Copyright (c) 2003-2023 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file LICENSE.
+ *
+ * @class CrossrefReferenceLinkingPlugin
+ * @brief Reference Linking plugin class
+ */
+
+namespace APP\plugins\generic\crossrefReferenceLinking;
+
+use APP\core\Application;
+use APP\facades\Repo;
+use APP\notification\NotificationManager;
+use APP\notification\Notification;
+use APP\publication\Publication;
+use APP\submission\Submission;
+use DOMDocument;
+use PKP\citation\CitationDAO;
+use PKP\config\Config;
+use PKP\context\Context;
+use PKP\core\JSONMessage;
+use PKP\db\DAORegistry;
+use PKP\linkAction\LinkAction;
+use PKP\linkAction\request\AjaxModal;
+use PKP\plugins\GenericPlugin;
+use PKP\plugins\Hook;
+use PKP\plugins\PluginRegistry;
+
+
+class CrossrefReferenceLinkingPlugin extends GenericPlugin {
+
+	public const CROSSREF_API_REFS_URL = 'https://doi.crossref.org/getResolvedRefs?doi=';
+
+	public const CROSSREF_API_REFS_URL_DEV = 'https://test.crossref.org/getResolvedRefs?doi=';
+
+	/**
+	 * @copydoc Plugin::register()
+	 */
+	public function register($category, $path, $mainContextId = null)
+	{
+		$success = parent::register($category, $path, $mainContextId);
+		if ($success && $this->getEnabled($mainContextId)) {
```

Comment:
Perhaps you can use early returns to decrease the indentation.

Reply:
I will leave this as it is, because everywhere else is the same...